### PR TITLE
add equation list to cartersian_graph

### PR DESCRIPTION
**Pull request by nvthblake**

### CODE REVIEW

**Comment by nvthblake** [comment]
@gwarchol Hey Greg, somehow the build failed because pubspec.yaml file is missing but I have the file on my branch, so I am not sure why the error showing up. Can you help me check? Thank you.

File: example/.dart_tool/package_config.json [hunk 1]
```diff
@@ -0,0 +1,170 @@
+{
```
Comment (gwarchol) [inline example/.dart_tool/package_config.json]: file seems unnecessary

File: example/.flutter-plugins [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# This is a generated file; do not edit or check into version control.
```
Comment (gwarchol) [inline example/.flutter-plugins]: should not be committed

File: example/.flutter-plugins-dependencies [hunk 3]
```diff
@@ -0,0 +1 @@
+{"info":"This is a generated file; do not edit or check into version control.","plugins":{"ios":[{"name":"cartesian_graph","path":"D:\\\\Users\\\\nvthblake\\\\Github\\\\cartesian_graph\\\\","dependencies":[]}],"android":[{"name":"cartesian_graph","path":"D:\\\\Users\\\\nvthblake\\\\Github\\\\cartesian_graph\\\\","dependencies":[]}],"macos":[],"linux":[],"windows":[],"web":[]},"dependencyGraph":[{"name":"cartesian_graph","dependencies":[]}],"date_created":"2020-11-08 10:33:18.337719","version":"1.22.1"}
```
Comment (gwarchol) [inline example/.flutter-plugins-dependencies]: should not be committed

File: example/.packages [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Generated by pub on 2020-11-08 10:33:17.986178.
```
Comment (gwarchol) [inline example/.packages]: should not be committed

File: example/pubspec.lock [hunk 5]
```diff
@@ -0,0 +1,182 @@
+# Generated by pub
```
Comment (gwarchol) [inline example/pubspec.lock]: should not be committed

File: lib/cartesian_graph.dart [hunk 6]
```diff
@@ -38,14 +39,16 @@ class CartesianGraph extends StatelessWidget{
...
 
-    if(equation != null){
-      AdvancedCalculator calculator = createCoordinateCalculator();
-      List<Coordinates> calculatedCoordinates = [];
-      for(double xCoordinate in display.xCoordinates){
-        double yCoordinate = calculator.calculateEquation(equation, xCoordinate);
-        calculatedCoordinates.add(Coordinates(xCoordinate, yCoordinate));
+    for (int i = 0; i < equations.length; i++) {
+      if (equations[i] != null){
+        AdvancedCalculator calculator = createCoordinateCalculator();
```
Comment (gwarchol) [inline lib/cartesian_graph.dart:44]: this should only occur once, not multiple times

File: pubspec.lock [hunk 7]
```diff
@@ -0,0 +1,364 @@
+# Generated by pub
```
Comment (gwarchol) [inline pubspec.lock]: should not be committed

File: lib/cartesian_graph.dart [hunk 8]
```diff
@@ -38,14 +39,16 @@ class CartesianGraph extends StatelessWidget{
       _plotCoordinates(display, builderCoordinates);
     }
 
-    if(equation != null){
-      AdvancedCalculator calculator = createCoordinateCalculator();
-      List<Coordinates> calculatedCoordinates = [];
-      for(double xCoordinate in display.xCoordinates){
-        double yCoordinate = calculator.calculateEquation(equation, xCoordinate);
-        calculatedCoordinates.add(Coordinates(xCoordinate, yCoordinate));
+    for (int i = 0; i < equations.length; i++) {
```
Comment (gwarchol) [inline lib/cartesian_graph.dart:42]: please add test for multiple equation scenario

**Review by gwarchol — CHANGES_REQUESTED** [review]

**Comment by gwarchol** [comment]
> 
> 
> @gwarchol Hey Greg, somehow the build failed because pubspec.yaml file is missing but I have the file on my branch, so I am not sure why the error showing up. Can you help me check? Thank you.

**Comment by gwarchol** [comment]
00:11 +1 -10: /home/runner/work/cartesian_graph/cartesian_graph/cartesian_graph/test/cartesian_graph_test.dart: Segment plotting plots equation display coordinates with equation [E] is the build failure

**Comment by gwarchol** [comment]
@nvthblake this is still missing tests

**Pull request merged**